src/org/labkey/test/components/glassLibrary/ParentEntityEditPanel.java

```diff
@@ -0,0 +1,475 @@
+        if(!typeCombo.getSelections().contains("Select a Source Type ...") &&
+                        !typeCombo.getSelections().contains("Select a Parent Type ..."))
+        {
+            // Since there is already a parent need to click the "Add" button to add a new one.
+            elementCache().addButton.click();
+
+            // Now need to go find the new combo but wait until it shows up.
+            WebDriverWrapper.waitFor(()-> numberOfTypeFields() > numOfTypes,
+                    "The new type field did not show up in a timely fashion.",
+                    1_000);
+
+            typeCombo = ReactSelect.finder(getDriver())
+                    .followingLabelWithSpan(_parentType.getType() + " Type " + (numOfTypes + 1))
+                    .find(this);
+
+        }
+
+        return typeCombo;
+
+    }
+
+    /**
+     * Get the type combos by it's index (zero based). This would include any 'add new' combos.
+     *
+     * @return A combo at the given position in the colection of combos..
+     */
+    protected ReactSelect getTypeCombo(int index)
+    {
+        return getAllTypeCombo().get(index);
+    }
+
+    /**
+     * Get the type combos by it's index (zero based). This would include any 'add new' combos.
+     *
+     * @return A combo at the given position in the colection of combos..
+     */
+    protected ReactSelect getTypeCombo(String labelText)
+    {
+        return ReactSelect.finder(getDriver())
+                .followingLabelWithSpan(labelText)
+                .find(this);
+    }
+
+    /**
+     * Get the type combos by it's index (zero based). This would include any 'add new' combos.
+     *
+     * @return A combo at the given position in the colection of combos..
+     */
+    protected ReactSelect getTypeComboWithSelection(String selection)
+    {
+        ReactSelect theCombo = null;
+        List<ReactSelect> allCombos = getAllTypeCombo();
+        for(ReactSelect combo : allCombos)
+        {
+            if(combo.getSelections().contains(selection))
+            {
+                theCombo = combo;
+                break;
+            }
+        }
+
+        return theCombo;
+    }
+
+    /**
+     * Get all of the type combos in the panel. This includes those already set and any 'add new' combos as well.
+     *
+     * @return A collection of all of the combos that select the type.
+     */
+    protected List<ReactSelect> getAllTypeCombo()
+    {
+        // Calling ReactSelect.finder(getDriver()).findAll(this) is not specific enough, it returns all divs with
+        // 'select' in the class. That can return multiple controls for one react select (div.Select, div.Select-control,
+        // div.Select-value,div.Select-input). That is if there is only one element selected. If it is a multi-select
+        // control each selected value will have div.Select-value.
+
+        List<ReactSelect> allControls = ReactSelect.finder(getDriver()).findAll(this);
```

Comment:
This sounds like a bug in the `ReactSelect`. You might be able to fix this problem somewhat by changing [`BaseReactSelect.Locators#selectContainer`](https://github.com/LabKey/testAutomation/blob/develop/src/org/labkey/test/components/glassLibrary/components/BaseReactSelect.java#L352) to use `tagWithClass` instead of `tagWithClassContaining`. Though that would get the multi-select id combos so it doesn't really help here.
You might be able to avoid all this rigamarole though by using your own `Locator`.
```
List<ReactSelect> typeCombos = ReactSelect.finder(getDriver())
        .locatedBy(Locator.tag("div").withClasses("Select", "Select--single"))
        .findAll(this);
```

Reply:
Good call! I changed the function to return this collection.